### PR TITLE
Feature/inventory app

### PR DESCRIPTION
**Pull request by kasenvr**
To test this app, constant rebuilding will be necessary, many functions can be tested and prototyped in the browser, however loading it into the interface is necessary for many cases. And ultimately, a final testing pass in the interface will be needed anyway to confirm it's working.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-33f1b9d_6c5dbb5b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-33f1b9d.tar_6c5dbb5b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-33f1b9d_6c5dbb5b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-33f1b9d.tar_6c5dbb5b.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-33f1b9d_6c5dbb5b.exe

File: scripts/system/inventory/public/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//  Created by kasenvr@gmail.com on 7 Apr 2020
+//  Copyright 2020 Vircadia and contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+-->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/public/index.html:14]: Indents should be 4 spaces.

File: scripts/system/inventory/src/main.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import Vue from 'vue'
+import App from './App.vue'
+import vuetify from './plugins/vuetify';
+import { store } from './plugins/store';
+
+Vue.config.productionTip = false;
+
+window.vm = new Vue({
+  vuetify,
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/main.js:19]: Indents should be 4 spaces.

File: scripts/system/inventory/vue.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+*/
+
+module.exports = {
+    publicPath: "./",
+    assetsDir: "./",
+    "transpileDependencies": [
+        "vuetify"
+    ]
+}
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/vue.config.js:17]: Should have a newline at the end of the file.

File: scripts/system/inventory/public/inventory.js [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    }
+
+    function initializeInventoryApp() {
+        sendSettings();
+        sendInventory();
+        sendReceivingItemQueue();
+    }
+
+    function onOpened() {
+        console.log("hello world!");
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/public/inventory.js:270]: Unwanted log file item?

File: scripts/system/inventory/public/inventory.js [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+    }
+
+    function onClosed() {
+        console.log("hello world!");
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/public/inventory.js:274]: Unwanted log file item?

File: scripts/system/inventory/src/App.vue [hunk 6]
```diff
@@ -0,0 +1,1430 @@
...
+    }
+}
+
+if (!browserDevelopment()) {
+    // eslint-disable-next-line
+    EventBridge.scriptEventReceived.connect(function(receivedCommand) {
+        receivedCommand = JSON.parse(receivedCommand);
+        // alert("RECEIVED COMMAND:" + receivedCommand.command)
+        if (receivedCommand.app === "inventory") {
+        // We route the data based on the command given.
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:706]: Should be indented a further 4 spaces.

File: scripts/system/inventory/src/App.vue [hunk 7]
```diff
@@ -0,0 +1,1430 @@
...
+            if (receivedCommand.command === 'script-to-web-inventory') {
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:707]: A `switch` statement could (optionally) be used in place of all these `if`s.

File: scripts/system/inventory/src/App.vue [hunk 8]
```diff
@@ -0,0 +1,1430 @@
...
+            {
+                name: "Is",
+                uuid: "{4131531653756756576543652562}",
+                distance: 3.23,
+            },
+            {
+                name: "This?",
+                uuid: "{4131531676575653652562}",
+                distance: 1,
+            },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:792]:
```suggestion
            }
```

File: scripts/system/inventory/src/App.vue [hunk 9]
```diff
@@ -0,0 +1,1430 @@
...
+        sortBy: "alphabetical",
+        settings: {
+            "displayDensity": {
+                "size": 1,
+                "labels": [
+                    "List",
+                    "Compact",
+                    "Large",
+                ],
+            },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:803]:
```suggestion
                ]
            }
```

File: scripts/system/inventory/src/App.vue [hunk 10]
```diff
@@ -0,0 +1,1430 @@
...
+                        "senderDistance": 1.22,
+                        "data": {
+                            "type": "script",
+                            "name": "REALLYLONGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG",
+                            "url": "https://butwhythough.com/looool.js",
+                            "uuid": "REALLYLONNGGGGGGGG",
+                        }
+                    }
+                ],
+            },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:773]:
```suggestion
                ]
            }
```

File: scripts/system/inventory/src/App.vue [hunk 11]
```diff
@@ -0,0 +1,1430 @@
...
+                    "List",
+                    "Compact",
+                    "Large",
+                ],
+            },
+        },
+        appVersion: "2.0.2",
+        darkTheme: true,
+        drawer: false,
+        disabledProp: true,
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:808]:
```suggestion
        disabledProp: true
```

File: scripts/system/inventory/src/App.vue [hunk 12]
```diff
@@ -0,0 +1,1430 @@
...
+            if (request === "edit") {
+                this.folderList = [
+                    {
+                        "name": "No Change",
+                        "uuid": "No Change"
+                    },
+                    {
+                        "name": "No Folder", 
+                        "uuid": "No Folder"
+                    },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:1078]:
```suggestion
                    }
```

File: scripts/system/inventory/src/App.vue [hunk 13]
```diff
@@ -0,0 +1,1430 @@
...
+                ];
+                
+                generateList = this.recursiveFolderPopulate(this.itemsStore, null);
+                
+            } else if (request === "add") {
+                this.folderList = [
+                    {
+                        "name": "No Folder", 
+                        "uuid": "No Folder"
+                    },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:1088]:
```suggestion
                    }
```

File: scripts/system/inventory/src/App.vue [hunk 14]
```diff
@@ -0,0 +1,1430 @@
...
+            } else if (request === "editFolder") {
+                this.folderList = [
+                    {
+                        "name": "No Change",
+                        "uuid": "No Change"
+                    },
+                    {
+                        "name": "No Folder", 
+                        "uuid": "No Folder"
+                    },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:1102]:
```suggestion
                    }
```

File: scripts/system/inventory/src/App.vue [hunk 15]
```diff
@@ -0,0 +1,1430 @@
...
+        },
+        recursiveFolderPopulate: function(indexToSearch, avoidFolder) {
+            for (var i = 0; i < indexToSearch.length; i++) {
+                if (Object.prototype.hasOwnProperty.call(indexToSearch[i], "items")) {
+                    // We want to avoid adding the folder itself and also any child folders it may have, putting a folder within its child folder will nuke it.
+                    if (avoidFolder !== indexToSearch[i].uuid) {
+                        // console.info("AvoidFolder", avoidFolder, "indexToSearch[i].uuid", indexToSearch[i].uuid);
+                        this.recursiveFolderHoldingList.push({
+                            "name": indexToSearch[i].name,
+                            "uuid": indexToSearch[i].uuid,
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:1190]:
```suggestion
                            "uuid": indexToSearch[i].uuid
```

File: scripts/system/inventory/src/App.vue [hunk 16]
```diff
@@ -0,0 +1,1430 @@
...
+        itemsStore: {
+            get() {
+                return this.$store.state.items;
+            },
+            set(value) {
+                this.$store.commit('mutate', {
+                    property: 'items', 
+                    with: value
+                });
+            },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/App.vue:1272]:
```suggestion
            }
```
Etc., below.

File: scripts/system/inventory/src/assets/styles/styles.css [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+    font-weight: lighter;
+}
+
+/* Universal */
+
+.handle {
+    background-color: rgba(0, 0, 0, 0.3);
+}
+
+.inventoryApp::-webkit-scrollbar { width: 0 !important }
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/assets/styles/styles.css:78]: Missing new line at end of file.

File: scripts/system/inventory/src/components/ItemIterator.vue [hunk 18]
```diff
@@ -0,0 +1,391 @@
...
+            </v-list-group>
+        </v-item-group>
+    </draggable>
+</template>
+
+
+<script>
+
+import { EventBus } from '../plugins/event-bus.js';
+import draggable from 'vuedraggable'
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/components/ItemIterator.vue:199]:
```suggestion
import draggable from 'vuedraggable';
```

File: scripts/system/inventory/src/components/ItemIterator.vue [hunk 19]
```diff
@@ -0,0 +1,391 @@
...
+        addDialogStore: {
+            get() {
+                return this.$store.state.addDialog;
+            },
+            set(value) {
+                this.$store.commit('mutate', {
+                    property: 'addDialog', 
+                    with: value
+                });
+            },
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/components/ItemIterator.vue:233]:
```suggestion
            }
```
Etc. below.

File: scripts/system/inventory/src/plugins/store.js [hunk 20]
```diff
@@ -0,0 +1,321 @@
...
+    devtools: true,
+    state: {
+        items: [
+            // This is test data and is primarily used for in browser development.
+            {
+                "type": "script",
+                "name": "VRGrabScale",
+                "url": "https://gooawefaweawfgle.com/vr.js",
+                "folder": "No Folder",
+                "uuid": "54254354353",
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/plugins/store.js:26]:
```suggestion
                "uuid": "54254354353"
```
Etc.

File: scripts/system/inventory/src/plugins/store.js [hunk 21]
```diff
@@ -0,0 +1,321 @@
...
+        moveFolder (state, payload) {
+            let { items } = state;
+            
+            if (payload.parentFolderUUID === "top") {
+                payload.findFolder.returnedItem.folder = "No Folder";
+                // console.info("Going to push...", payload.findFolder.returnedItem);
+                // console.info("Containing these items...", payload.findFolder.returnedItem.items);
+
+                items.push(payload.findFolder.returnedItem);
+                Vue.set(state,'items', items);
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/plugins/store.js:287]:
```suggestion
                Vue.set(state, 'items', items);
```
Etc. below.

File: scripts/system/inventory/src/plugins/store.js [hunk 22]
```diff
@@ -0,0 +1,321 @@
...
+                // console.info("Containing these items...", payload.findFolder.returnedItem.items);
+                // console.info("Into...", payload.findParentFolder.returnedItem);
+
+                payload.findItem.returnedItem.folder = payload.findParentFolder.name;
+                payload.findParentFolder.returnedItem.items.push(payload.findItem.returnedItem);
+                Vue.set(state,'items', items);
+            }
+        }
+    }
+})
```
Comment (ctrlaltdavid) [inline scripts/system/inventory/src/plugins/store.js:321]: Missing new line at end of file.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-5934eea_087ede2f.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-5934eea.tar_087ede2f.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-5934eea.tar_087ede2f.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-5934eea_087ede2f.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-edac00d_e5738427.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-edac00d.tar_e5738427.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-edac00d.tar_e5738427.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-edac00d_e5738427.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR521/Vircadia-Alpha-PR521-edac00d_e5738427.exe

**Pull request merged**